### PR TITLE
Release geminabox version

### PR DESCRIPTION
**Pull request by rafalbigaj**
Allow feature versions of geminabox (current 0.12.4)

### CODE REVIEW

**Comment by bleything** [comment]
This looks fine but I don't have a geminabox server to test against. I asked @halostatue to confirm it works fine, at which point I'll merge and release.

Also I recognize that this PR is over a year old, and you've probably long since figured out something else. Sorry for that.

**Comment by halostatue** [comment]
``` diff
diff --git 1/Rakefile 2/Rakefile.alt
index 27bd624..fb40eb7 100644
--- 1/Rakefile
+++ 2/Rakefile.alt
@@ -3,12 +3,19 @@ require 'hoe'

 Hoe.plugin :doofus, :git

+require 'pathname'
+config = Pathname('~/.gem/geminabox').expand_path
+geminabox = YAML.load(config.read)[:host] if config.exist?
+
+Hoe.plugin :geminabox
+
 Hoe.spec 'hoe-geminabox' do
   developer 'Ben Bleything', 'ben@bleything.net'
   ### Use markdown for changelog and readme
   self.history_file = 'CHANGELOG.md'
   self.readme_file  = 'README.md'

+  self.geminabox_server = geminabox
+
   ### dependencies!
   self.extra_deps     << ['geminabox', '>= 0.3.3']
 end
```

With these modifications applied as `Rakefile.alt`, I then ran `rake -f Rakefile.alt release VERSION=0.3.1` and the release worked.

It has been possible to get hoe-geminabox to work _most_ of the time without this change, but this change will guarantee that hoe-geminabox works _all_ of the time. The case where it failed is a fresh install when you have a geminabox server behind HTTPS (as I have set up through OpenShift). The version of geminabox that you are currently locked to doesn’t support HTTPS, so you get difficult-to-understand failures related to inability to resolve the HTTP URL or cert failures because you are asking for HTTP but getting HTTPS.

If you remembered this was the cause, you could manually run `gem install geminabox` to install a later version of `geminabox` and everything would Just Work because you do the right thing and run `sh %Q(gem inabox …)`. Unless…

…you were using bundler. Because bundler locks you to the versions it knows about, you don’t get an opportunity to see the newer version of `geminabox`.

There are a couple of minor quibbles I would have; geminabox itself is 0.12.4 now, which suggests that between this version and 1.0 if they ever release one…the interface _could_ change, which means that the `>= 0.3.3` specification is invalid because 1.0 _might_ be incompatible (unlikely, but…). This could be fixed by using slightly pessimistic versioning (`~> 0.3`).

It also doesn’t help people who are having this problem _now_ but aren’t sure what the problem is (and that it’s caused by using such an old version of geminabox). To make this safer, you could specify `~> 0.4` or even `~> 0.12`, _but_ this introduces a different problem in that I don’t know which version of geminabox introduced _working_ HTTPS support, which would be the ideal minimum version

TL;DR: this works. There are other considerations, but you can definitely merge and release this as it is now safe, even under Bundler.

**Comment by bleything** [comment]
Thank you both. I've applied @rafalbigaj's changes and will be tweaking a few things to match @halostatue's suggestions and releasing a new version in the next hour or so.

**Comment by bleything** [comment]
Released as 0.4.0.
